assets/content/cookbook/Beginner/string-variable-reflection.md

```diff
@@ -0,0 +1,38 @@
+[tags]: / "beginner,other"
+
+# Using string as variable (with reflection)
+
+Extract variable name from string and using it as variable
+
+## Implementation
+```haxe
+class ReflectionTest {
+  @:keep var ref : String = "This is Reflection Test";
+  
+  public function show() 
+  {
+    var t : String = Reflect.field(this,"ref");
+    trace(t);
+  }
+	
+  public function new()
+  {
+  }
+}
+```
+
+## Usage
+```haxe
+class Main {
+  static public function main():Void {
+    var r : ReflectionTest = new ReflectionTest();
+	r.show();
+  }
+}
+```
+
+Haxe have dead-code elimination enabled by default.In the example, the variable 'ref' is not referenced anywhere, so it will be removed. We need to add @:keep to that field, because it's only referenced by reflection.
```

Comment:
Just a little touch-up is necessary. My suggestion follows:

Haxe has dead code elimination (DCE) enabled by default. This compiler feature identifies and eliminates all unused code during compilation. In the example above, the variable `ref` is referenced only through reflection and not directly. Because of that, it will be marked for removal by DCE. To keep it from being eliminated, we need to add `@:keep` compiler metadata to the `ref` field.